### PR TITLE
fix: process multiple host entries

### PR DESCRIPTION
**Pull request by CertainLach**
PR https://github.com/Eugeny/russh/pull/306 broke russh config parsing for me.

My ssh config looks roughly like this:

```
Host *
    Common settings
Host wanted
   Host-specific settings
```

Before https://github.com/Eugeny/russh/pull/306 common settings were ignored, but after - common settings were applied, yet config parser stops right on next `Host` section start, so settings I need were ignored instead.

This PR brings behavior closer to what openssh itself does - for requested host, it applies all the settings with maching glob, not only the first one.

### CODE REVIEW

File: russh-config/src/lib.rs [hunk 1]
```diff
@@ -166,22 +171,10 @@ pub fn parse(file: &str, host: &str) -> Result<Config, Error> {
...
-                config = Some(c)
             }
         }
     }
-    if let Some(config) = config {
-        Ok(config)
-    } else {
-        Err(Error::HostNotFound)
-    }
+    Ok(config)
```
Comment (CertainLach) [inline russh-config/src/lib.rs:177]: With glob directives, it makes little sense to return HostNotFound error, I may return this error, but I think most of the time users just want to parse all relevant host settings, and will `unwrap_or_default` anyways after config parse call.

Comment (Eugeny) [reply russh-config/src/lib.rs:177]: Makes sense

**Comment by Eugeny** [comment]
Thank you!

**Comment by Eugeny** [comment]
@all-contributors please add @CertainLach for code

**Pull request merged**